src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -293,4 +317,45 @@ class BlobCacheServiceImpl implements BlobCacheService {
             }
         }
     }
+
+   /**
+    * get the size of the blob stored in the cache
+    *
+    * @return {@link Long} the size of the blob stored in the cache
+    */
+   protected Long getBlobSize(String key) {
+        try {
+            final headObjectRequest =
+                    HeadObjectRequest.builder()
+                            .bucket(blobConfig.storageBucket.replaceFirst("s3://",""))
+                            .key(key)
```

Comment:
This is is to tight to s3, we may want to use other storage in the future. Let's use [BucketTokenizer](https://github.com/seqeralabs/wave/blob/67cbd9da631bd9336a7f457208821cb08f9c6666/src/main/groovy/io/seqera/wave/util/BucketTokenizer.groovy#L38-L38)


Same for delete method below

Reply:
done